### DISCUSSION TITLE
How can I visualize the output of analyze command in github actions

### DISCUSSION THREAD

**Discussion by kamil-oberaj** [JosefPihrt/Roslynator#1121] [General]
I'd like to show analyze command summary (by using --output) in my github action summary, Do You have any ideas how to visualize/view and beautify the output?

**Comment by josefpihrt** [comment]
Hi,

so you want something better than standard output, for some reason, am I right?

If you are asking if it's possible to visualize xml file in other form than standard workflow text output then I have no idea if it's possible, sorry.

If you are asking whether it's possible to use xml output, parse it and display text output in other way than standard output then it's definitely possible by writing some script/program that will be executed after `analyze` command. You would probably want to use `-v q` parameter to suppress standard output. And also you would have to use `continue-on-error` to enable continuation of workflow.

**Reply by kamil-oberaj** [reply]
Hi, thank You for Your reply.

Maybe it's not that I want/must have that functionality in Your package. I was just wondering if there is any way to display the xml output in action summary. By displaying I mean something that'll format the xml to some table with contents or expandable list. Maybe just summary, I don't really have that strong preference there.

I was looking for other actions/scripts that support Your xml output, but I couldn't find one. This is the reason for this thread.

Edit:
I was kinda surprised by failed workflow when the action failed, even if the full analyse was run. I've found the solution and it is, as You've said, `continue-on-error`

**Comment by josefpihrt** [comment]
Yes, currently there are no scripts/actions that would process xml output. You can contribute with the script that would be put into `tools` folder to be available for others.

As an author, I'm curious what is missing/wrong with the standard output or how would you improve it. The output can be improved if we agree.

**Reply by kamil-oberaj** [reply]
Just to clarify: There is nothing wrong with the output 😃 

I've never written any of scripts formatting xml files, like this, but I'll message You if something comes to my mind.
BTW: Thank You for Your solution with roslyn! It helped me a lot!

**Comment by josefpihrt** [comment]
I'm glad you like Roslynator! 😀
